tests/test_writer.py

```diff
@@ -1559,16 +1567,17 @@ def test_watermark():
 
 
 @pytest.mark.enable_socket()
-@pytest.mark.timeout(4)  # this was a lot slower before PR #2086
```

Comment:
I think that one is the cleaner option. If we change it, it should be a separate PR (but I'm uncertain if we should change it at all).

You mentioned that you had issues with that test. Was it local or in CI? What exactly was the issue / the error message?

Reply:
I agree it is not the good place, but wanted to discuss about.

The issue during the checks was just it failed with the timeout. It is not the first time this is why I've proposed this change.
I consider the time is ok but for me it is just because We are measuring too many instructions. That' why I've focused on the  specif code of the merge.
If you like it just give me a go and I will generate a dedicated PR dealing with this change only